pkg/securitypolicy/securitypolicy_linux.go

```diff
@@ -18,6 +20,114 @@ import (
 //nolint:unused
 const osType = "linux"
 
+// validateHostData fetches SNP report (if applicable) and validates `hostData` against
+// HostData set at UVM launch.
+func validateHostData(hostData []byte) error {
```

Comment:
This code and the code in the windows version are functionally equivalent. I suspect the main difference is the amdsevsnp vs PsPDriver as the owner of the PSP device. Can we get that behind an abstraction so this code is actually the same concrete function?

Reply:
They are already abstracted and call their respective driver functions though.